### PR TITLE
Add consul segment fingerprint

### PR DESCRIPTION
**Pull request by daktari**
This PR allows to add consul segment attribute to nomad clients.

BR

### CODE REVIEW

**Comment by daktari** [comment]
Hi! Any updates about this ?

File: client/fingerprint/consul.go [hunk 1]
```diff
@@ -81,6 +81,11 @@ func (f *ConsulFingerprint) Fingerprint(req *FingerprintRequest, resp *Fingerpri
 	} else {
 		f.logger.Warn("unable to fingerprint consul.datacenter")
 	}
+	if g, ok := info["Member"]["Tags"]["segment"].(string); ok {
```
Comment (notnoop) [inline client/fingerprint/consul.go:84]: This has a compile error now - update and we can merge.  Thanks!

**Comment by daktari** [comment]
Ok! Updated

**Comment by tgross** [comment]
Looks like there's still a compile error on this:

```
==> Linting source code...
WARN [runner] Can't run linter goanalysis_metalinter: ineffassign: analysis skipped: errors in package: [/go/src/github.com/hashicorp/nomad/client/fingerprint/consul.go:84:14: invalid operation: cannot index info["Member"]["Tags"] (map index expression of type interface{})] 
ERRO Running error: ineffassign: analysis skipped: errors in package: [/go/src/github.com/hashicorp/nomad/client/fingerprint/consul.go:84:14: invalid operation: cannot index info["Member"]["Tags"] (map index expression of type interface{})] 
```

**Review by tgross — APPROVED** [review]

**Comment by tgross** [comment]
I hadn't noticed that new commits came in to get this green, thanks for those @daktari and sorry about the delay. LGTM!

**Pull request merged**